packages/contracts-bedrock/contracts/L1/L2OutputOracleV2.sol

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import { SafeCall } from "../libraries/SafeCall.sol";
+import { Semver } from "../universal/Semver.sol";
+import { Types } from "../libraries/Types.sol";
+
+/**
+ * @custom:proxied
+ * @title L2OutputOracleV2
+ * @notice The L2OutputOracleV2 contains a map of L2 block numbers to their associated
+ *         L2 state outputs, where each output is a commitment to the state of the L2 chain.
+ *         Other contracts like the OptimismPortal use these outputs to verify information
+ *         about the state of L2.
+ */
+contract L2OutputOracleV2 is Initializable, Semver {
+    /**
+     * @notice The time between L2 blocks in seconds. Once set, this value MUST NOT be modified.
+     */
+    uint256 public immutable L2_BLOCK_TIME;
+
+    /**
+     * @notice The address of the challenger. Can be updated via upgrade.
+     */
+    address public immutable CHALLENGER;
+
+    /**
+     * @notice Minimum time (in seconds) that must elapse before a withdrawal can be finalized.
+     */
+    uint256 public immutable FINALIZATION_PERIOD_SECONDS;
+
+    /**
+     * @notice Minimum cost of submitting an output proposal.
+     */
+    uint256 public immutable MINIMUM_OUTPUT_PROPOSAL_COST;
```

Comment:
Curious why this is not more easily modifiable? It seems like something hard to parametrize properly that we may wish to update.

Reply:
I realize this probably ought to be part of the `BondManager` anyways :)